scripts/generate-java-code-coverage.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+JACOCO_VERSION=0.8.12
+
+if [ "$(basename "$(pwd)")" != "java" ]; then
+    echo "This script must be run from the 'java' directory."
+    exit 1
+fi
+
+# make -C ../cpp srcs
+# make
+
+if [ ! -f org.jacoco.cli-nodeps.jar ]; then
+    wget https://repo1.maven.org/maven2/org/jacoco/org.jacoco.cli/$JACOCO_VERSION/org.jacoco.cli-$JACOCO_VERSION-nodeps.jar -O org.jacoco.cli-nodeps.jar
+fi
+
+if [ ! -f org.jacoco.agent-runtime.jar ]; then
+    wget https://repo1.maven.org/maven2/org/jacoco/org.jacoco.agent/$JACOCO_VERSION/org.jacoco.agent-$JACOCO_VERSION-runtime.jar -O org.jacoco.agent-runtime.jar
+fi
+
+python3 allTests.py --all --workers=4 --debug --jacoco $(pwd)/org.jacoco.agent-runtime.jar
+
+# We don't want any coverage for IceGridGUI
+find src/IceGridGUI -name '*.java' -exec rm {} \;
+
+# We don't want any coverage for generated code. We remove their class files and any inner classes.
+jacocoArgs=()
+for dir in src/*; do
+    [ -d "$dir" ] || continue
```

Comment:
Actually, I think the most correct solution is to just `continue` when `dir == IceGridGui` or whatever.
This means we don't need to `find rm` anything, which would be nice.

Also, the current approach feels a little weird. `find src/IceGridGUI -name '*.java' -exec rm {} \;` removes all the source files, but not the _class_ files. So, we're still passing all the `IceGridGUI` classes to jacoco. But, I guess it doesn't display any output for them because it doesn't know about the source files?
Feels sketchy. It'd be better to just not pass the `IceGridGui` directory or anything at all IMO.